model/task_queue_service_dependency.go

```diff
@@ -574,3 +550,27 @@ func (d *basicCachedDAGDispatcherImpl) nextTaskGroupTask(unit schedulableUnit) *
 
 	return nil
 }
+
+func checkUnmarkedBlockingTasks(t *task.Task, dependencyCaches map[string]task.Task) error {
+	catcher := grip.NewBasicCatcher()
```

Comment:
what's the reason for using the catcher here instead of just erroring at the point of error?

Reply:
if the blocked check fails the activated check is still useful.